src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/resourcetakes/view/ResourceTakesView.kt

```diff
@@ -0,0 +1,70 @@
+package org.wycliffeassociates.otter.jvm.app.ui.resourcetakes.view
+
+import javafx.collections.ListChangeListener
+import org.wycliffeassociates.controls.ChromeableTabPane
+import org.wycliffeassociates.otter.common.data.model.ContentType
+import org.wycliffeassociates.otter.common.domain.content.Recordable
+import org.wycliffeassociates.otter.jvm.app.ui.resourcetakes.viewmodel.TakesViewModel
+import java.util.EnumMap
+import tornadofx.*
+import org.wycliffeassociates.otter.jvm.utils.getNotNull
+
+class ResourceTakesView : View() {
+    private val viewModel: TakesViewModel by inject()
+    private val tabPane = ChromeableTabPane()
+
+    class ContentTypeToTabMap(map: Map<ContentType, TakesTab?>): EnumMap<ContentType, TakesTab>(map)
+    private val contentTypeToTabMap = ContentTypeToTabMap(
```

Comment:
Subclassing doesn't seem necessary. Can't contentTypeToTabMap just be an EnumMap?

Reply:
Actually sorry I meant to comment on this. In MapUtils.kt, `getNotNull()` uses reflection to get the class name when throwing an exception. That's really the only reason I subclassed EnumMap.

Reply:
I might consider just making getNotNull() a method of ContentTypeToTabMap. I'm ok with this as-is, though.

Reply:
can we avoid nulls altogether here?

Reply:
Not quite, we can't guarantee that a key exists in the map